File: backend/app/schemas/job.py
from pydantic import BaseModel


class JobMetadata(BaseModel):
    job_id: str
    status: str
    progress: int = 0
    current_step: str | None = None
    status_message: str | None = None
    pages_processed: int | None = None
    total_pages: int | None = None
    elapsed_seconds: float | None = None
    eta_seconds: int | None = None
    eta_formatted: str | None = None
    processing_speed_pps: float | None = None
    error: str | None = None
    marketplace: str | None = None
    original_filename: str | None = None
    stored_filename: str | None = None
    file_size: int | None = None
    page_count: int | None = None
    label_groups: int | None = None
    uploaded_filenames: list[str] | None = None
    uploaded_file_count: int | None = None
    mime_type: str | None = None
    created_at: str | None = None
    updated_at: str | None = None

File: frontend/src/types/job.ts
export interface JobMetadata {
  job_id: string;
  status: string;
  progress?: number;
  current_step?: string;
  pages_processed?: number;
  total_pages?: number;
  error?: string | null;
  marketplace?: string | null;
  filename?: string | null;
  stored_filename?: string | null;
  file_size?: number | null;
  page_count?: number | null;
  label_groups?: number | null;
  uploaded_filenames?: string[];
  uploaded_file_count?: number;
  mime_type?: string;
  created_at: string;
  updated_at: string;
}

export type JobWorkflowStage = 
  | 'upload_completed'
  | 'marketplace_detected'
  | 'labels_extracted'
  | 'analysis_ready'
  | 'statistics_generated'
  | 'waiting_for_sort'
  | 'processing'
  | 'completed'
  | 'error';

export interface UploadResultData {
  job_id: string;
  status: string;
  marketplace?: string;
  page_count?: number;
  label_count?: number;
  uploaded_file_count?: number;
  file_size?: number;
}
